python/lsst/afw/table/catalogMatches.py

```diff
@@ -0,0 +1,164 @@
+
+    for rFrom, rTo in zip(catalog, target):
+        for kFrom, kTo in zip(sourceKeys, targetKeys):
+            try:
+                rTo.set(kTo, rFrom.get(kFrom))
+            except:
+                print "Error setting: %s %s %s %s" % (type(rFrom), type(rTo), type(kFrom), type(kTo))
+
+    return target
+
+def matchesToCatalog(matches, matchMeta):
+    """Denormalise matches into a Catalog of "unpacked matches"
+
+    \param[in] matches    unpacked matches
+    \param[in] matchMeta  metadata for matches (must have .add attribute)
+
+    \return  lsst.afw.table.BaseCatalog of matches (with ref_ and src_ prefix identifiers
+             for referece and source entries, respectively)
+    """
+
+    if len(matches) == 0:
+        raise RuntimeError("No matches provided.")
+
+    refSchema = matches[0].first.getSchema()
+    srcSchema = matches[0].second.getSchema()
+
+    mergedSchema = copySchema(refSchema, Schema(), targetPrefix="ref_")
+    mergedSchema = copySchema(srcSchema, mergedSchema, targetPrefix="src_")
+    distKey = mergedSchema.addField("distance", type=float, doc="Distance between ref and src")
+
+    mergedCatalog = BaseCatalog(mergedSchema)
+    copyCatalog([m.first for m in matches], mergedCatalog, sourceSchema=refSchema, targetPrefix="ref_")
+    copyCatalog([m.second for m in matches], mergedCatalog, sourceSchema=srcSchema, targetPrefix="src_")
+    for m, r in zip(matches, mergedCatalog):
+        r.set(distKey, m.distance)
+
+    # obtaining reference catalog name if one is setup
+    try:
+        catalogName = os.path.basename(getPackageDir("astrometry_net_data"))
+    except:
+        catalogName = 'NOT_SET'
+    matchMeta.add('REFCAT', catalogName)
+    mergedCatalog.getTable().setMetadata(matchMeta)
+
+    return mergedCatalog
+
+def matchesFromCatalog(catalog, sourceSlotConfig=None, prefix=""):
+    """Generate a list of ReferenceMatches from a Catalog of "unpacked matches"
+
+    \param[in] catalog           catalog of matches.  Must have schema where reference entries are
+                                 prefixed with "ref_" and source entries are prefixed with "src_"
+    \param[in] sourceSlotConfig  configuration for source slots (optional)
+    \param[in] prefix            prefix for slot config setup (optional)
+
+    \returns   lsst.afw.table.ReferenceMatch of matches
+    """
+    if catalog is None:
+        # There are none
+        return []
+    refSchema = copySchema(catalog.schema, SimpleTable.makeMinimalSchema(), sourcePrefix="ref_")
+    refCatalog = SimpleCatalog(refSchema)
+    copyCatalog(catalog, refCatalog, sourcePrefix="ref_")
+
+    srcSchema = copySchema(catalog.schema, SourceTable.makeMinimalSchema(), sourcePrefix="src_")
+    srcCatalog = SourceCatalog(srcSchema)
+    copyCatalog(catalog, srcCatalog, sourcePrefix="src_")
+
+    if sourceSlotConfig is not None:
+        sourceSlotConfig.setupTable(srcCatalog.table, prefix=prefix)
```

Comment:
What is `sourcesSlotConfig`? I searched both [Doxygen](https://lsst-web.ncsa.illinois.edu/doxygen/x_masterDoxyDoc/search.php?query=setupTable) and [GitHub](https://github.com/search?utf8=✓&q=user%3Alsst+setupTable) for a `setupTable()` method, and came up blank.